graphql_compiler/query_pagination/parameter_generator.py

```diff
@@ -1,4 +1,172 @@
+                             .format(vertex_partition))
+
+    # Since we can't be sure the minimum observed value is the
+    # actual minimum value, we treat values less than it as part
+    # of the first quantile. That's why we drop the minimum and
+    # maximum observed values from the quantile list.
+    proper_quantiles = quantiles[1:-1]
+
+    # Get the relevant quantiles (ones inside the field_value_interval)
+    # TODO(bojanserafimov): It's possible that the planner thought there are enough quantiles
+    #                       to paginate, but didn't notice that there are filters that restrict
+    #                       the range of values into a range for which there are not enough
+    #                       quantiles. In this case, the pagination plan is not fully realized.
+    #                       The generated query will have fewer pages than the plan specified.
+    #
+    #                       One solution is to push all the pagination capacity logic
+    #                       into the cost estimator, and make it return along with the
+    #                       cardinality estimate some other metadata that the paginator would
+    #                       rely on.
+    min_quantile = 0
+    max_quantile = len(proper_quantiles)
+    if field_value_interval.lower_bound is not None:
+        min_quantile = bisect.bisect_left(proper_quantiles, field_value_interval.lower_bound)
+    if field_value_interval.upper_bound is not None:
+        max_quantile = bisect.bisect_left(proper_quantiles, field_value_interval.upper_bound)
+    relevant_quantiles = proper_quantiles[min_quantile:max_quantile]
+
+    # TODO(bojanserafimov): We deduplicate the results to make sure we don't generate pages
+    #                       that are known to be empty. This can cause the number of generated
+    #                       pages to be less than the desired number of pages.
+    return _deduplicate_sorted_generator(
+        proper_quantiles[index]
+        for index in _sum_partition(
+            len(relevant_quantiles) + 1,
+            vertex_partition.number_of_splits
+        )
+    )
+
+
+def generate_parameters_for_vertex_partition(schema_info, query_ast, parameters, vertex_partition):
+    """Return a generator of parameter values that realize the vertex partition.
+
+    This function returns a generator of (vertex_partition.number_of_splits - 1) values that
+    split the values at vertex_partition.pagination_field into vertex_partition.number_of_splits
+    almost equal chunks.
+
+    HACK(bojanserafimov): In some cases the number values returned is less than
+                          (vertex_partition.number_of_splits - 1). This happens when the pagination
+                          planner generates an impossible plan. This is obviously not desirable.
+                          This flaw is not visible to the end user, but shold be fixed ASAP.
```

Comment:
@obi1kenobi If this is acceptable for now, I can quickly polish test and merge this PR, and move on to the next 2-3 PRs (at least 100 lines each, not including tests) so we can get pagination until Feb 1.

Legit fix I can continue on after pagination works: https://github.com/kensho-technologies/graphql-compiler/pull/738

Reply:
Sure, that's fine. Maybe just add a link to #738 in this note so we know when to remove it.